lib/modules/hosts/youtube.js

```diff
@@ -11,49 +12,48 @@ export default {
 			(/^https?:\/\/(?:youtu\.be|(?:www\.|m\.)?youtube\.com)\/?(?:watch|embed)?(?:.*?v=|v\/|\/)([\w-]+)/i).exec(href);
 	},
 	handleLink(elem, [, hash]) {
-		// Check url for timecode e.g t=1h23m15s
-		const starttimeRe = /[&\?](?:t=|start=)([0-9hms]+)/i;
-		const starttimeResult = starttimeRe.exec(elem.href);
-		// End time is always in integer format, no need to parse
-		const endtimeRe = /[&\?]end=([0-9]+)/i;
-		const endtimeResult = endtimeRe.exec(elem.href);
-		// Finds playlist id if included
-		const playlistRe = /[&\?]list=([\w\-_]+)/i;
-		const playlistResult = playlistRe.exec(elem.href);
+		let src = `https://www.youtube.com/embed/${hash}?enablejsapi=1&enablecastapi=1`;
+		const paramResults = getUrlParams(elem.href);
+		for (const paramName in paramResults) {
```

Comment:
@erikdesjardins Oh, didn't think to do that. I'm only used to working in strong typed languages. Not a fan of doing things weakly typed and find switch cases similar than if cases in cases like this. Curious to know if there is actually any performance benefit, though. But I'll make the change.